src/server/webapi/objects.cpp

```diff
@@ -227,27 +227,30 @@ int H_ObjectDetails(Context *context)
 /**
  * Check if an object has any descendants matching the class filter
  */
-bool HasDescendantsMatchingClassFilter(uint32_t objectId, uint32_t userId, const std::unordered_set<int> &classFilter, std::unordered_set<uint32_t> &visited, int maxDepth, int currentDepth)
+static bool HasDescendantsMatchingClassFilter(uint32_t objectId, uint32_t userId, const std::unordered_set<int> &classFilter, std::unordered_set<uint32_t> &visited, int maxDepth, int currentDepth)
 {
    // Prevent infinite recursion and respect maximum depth
    if (visited.count(objectId) > 0 || currentDepth >= maxDepth)
       return false;
 
    visited.insert(objectId);
 
-   unique_ptr<SharedObjectArray<NetObj>> children = g_idxObjectById.getObjects(
-      [userId, objectId] (NetObj *obj) -> bool
-      {
-         if (obj->isHidden() || obj->isSystem() || obj->isDeleted() || !obj->checkAccessRights(userId, OBJECT_ACCESS_READ))
-            return false;
-         return obj->isDirectParent(objectId);
-      });
+   shared_ptr<NetObj> object = FindObjectById(objectId);
```

Comment:
caller of this function already has pointer to object. It can be passed as argument instead of passing ID and finding object.